core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -417,12 +426,17 @@ private void acquireNewPageIfNecessary(int required) {
   public void insertRecord(Object recordBase, long recordOffset, int length, int partitionId)
     throws IOException {
 
-    // for tests
     assert(inMemSorter != null);
     if (inMemSorter.numRecords() >= numElementsForSpillThreshold) {
-      logger.info("Spilling data because number of spilledRecords crossed the threshold {}" +
+      logger.info("Spilling data because number of spilledRecords ({}) crossed the threshold {}",
+        MDC.of(LogKeys.NUM_ELEMENTS_SPILL_RECORDS$.MODULE$,inMemSorter.numRecords()),
         MDC.of(LogKeys.NUM_ELEMENTS_SPILL_THRESHOLD$.MODULE$, numElementsForSpillThreshold));
       spill();
+    } else if (inMemRecordsSize >= recordsSizeForSpillThreshold) {
+      logger.info("Spilling data because size of spilledRecords ({}) crossed the threshold {}",
+        MDC.of(LogKeys.NUM_ELEMENTS_SPILL_RECORDS$.MODULE$,inMemSorter.numRecords()),
```

Comment:
```suggestion
        MDC.of(LogKeys.NUM_ELEMENTS_SPILL_RECORDS$.MODULE$,inMemRecordsSize),
```

Reply:
And even the log message and log key should be updated.

Reply:
Thanks. Fixed.